### PR TITLE
String tests

### PR DESCRIPTION
**Pull request by nnnnblaser**

### CODE REVIEW

File: builtins/match.go [hunk 1]
```diff
@@ -30,6 +30,15 @@ func match(a, b runtime.Object) (bool, error) {
 		default:
 			return matchRecordOf(a.(*runtime.List), b)
 		}
+	case *runtime.String:
+		if c := a.(*runtime.String); c.Len() == 1 && b.Len() == 1 {
+			if b.Value[0] == rune('?') || b.Value[0] == rune('*') || b.Value[0] == c.Value[0] {
```
Comment (5nord) [inline builtins/match.go:35]: A comment why this extra comparison is needed would be quite useful.

File: builtins/match.go [hunk 2]
```diff
@@ -30,6 +30,15 @@ func match(a, b runtime.Object) (bool, error) {
...
+				return true, nil
+			}
+			return false, runtime.Errorf("tried to match %d with %d", c.Value[0], b.Value[0])
+
+		}
+		return matchRecordOf(a.(*runtime.String), b)
```
Comment (5nord) [inline builtins/match.go:41]: I am not sure, but I think you can drop the type assertion (`.(*runtime.String)`).

File: builtins/match.go [hunk 3]
```diff
@@ -57,7 +66,7 @@ func matchRecordOf(val, pat sliceHolder) (bool, error) {
 	i, back_i := 0, -1
 	j, back_j := 0, -1
 	for i < val.Len() && j < pat.Len() {
-		if pat.Get(j) == runtime.AnyOrNone {
+		if pat.Get(j) == runtime.AnyOrNone || pat.Get(j).Equal(runtime.NewString("*")) {
```
Comment (5nord) [inline builtins/match.go:69]: This is also a good place for a explanatory comment, why this Equal call is required.

File: builtins/match.go (hunk 2 above)
Comment (nnnnblaser) [reply builtins/match.go:43]: You can't

**Pull request merged**